include/world_builder/topography/uniform.h

```diff
@@ -0,0 +1,85 @@
+/*
+  Copyright (C) 2018-2024 by the authors of the World Builder code.
+
+  This file is part of the World Builder.
+
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU Lesser General Public License as published
+   by the Free Software Foundation, either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public License
+   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef WORLD_BUILDER_TOPOGRAPHY_MODEL_UNIFORM_H
+#define WORLD_BUILDER_TOPOGRAPHY_MODEL_UNIFORM_H
+
+#include "world_builder/topography/interface.h"
+
+#include "world_builder/utilities.h"
+
+
+namespace WorldBuilder
+{
+
+  namespace TopographyModel
+  {
+    /**
+     * Register header file
+     */
+    //WB_REGISTER_COORDINATE_SYSTEM_HEADER(Uniform)
+
+
+    /**
+     * This implements a Uniform geometry model.The Uniform geometry model
+     * doesn't do anything with the coordinates, but is needed to have a common
+     * interface for all the geometry models.
+     */
+    class Uniform final : public Interface
+    {
+      public:
+        /**
+         * constructor
+         */
+        Uniform(WorldBuilder::World *world);
+
+        /**
+         * Destructor
+         */
+        ~Uniform() override final;
+
+        /**
+         * declare and read in the world builder file into the parameters class
+         */
+        static
+        void declare_entries(Parameters &prm, const std::string &parent_name = "");
+
+        /**
+         * declare and read in the world builder file into the parameters class
+         */
+        void parse_entries(Parameters &prm) override final;
+
+        /**
+         * declare and read in the world builder file into the parameters class
+         */
+        std::string
+        get_surface_type(Parameters &prm) override final;
+
+      private:
+
+        /**
+         * Whether to consider a reference or deformed surface.
+         */
+        const std::string  surface_type = "reference_surface";
```

Comment:
while a string is flexible, it is also quite expensive to use. In addition it could contain arbitrary new values. I would suggest to introduce an `enum` type in the public part of this class (or even outside the class, but inside the `Topography` namespace. Something like:

```
  enum SurfaceType
  {
    reference_surface,
    deformed_surface};
```

This also guarantees that only these two values can be used.